File: BE/CapstoneProject_SP25_IPAS/CapstoneProject_SP25_IPAS_BussinessObject/Entities/Report.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CapstoneProject_SP25_IPAS_BussinessObject.Entities
{
    public partial class Report
    {
        public int ReportID { get; set; }
        public string? ReportCode { get; set; }
        public string? QuestionOfUser { get; set; }
        public string? AnswerFromExpert { get; set; }
        public string? Description { get; set; }
        public string? ImageURL { get; set; }
        public DateTime? CreatedDate { get; set; }
        public bool? IsTrainned { get; set; }
        public int? AnswererID { get; set; }
        public int? QuestionerID { get; set ; }

        public virtual User? Answerer { get; set; }
        public virtual User? Questioner { get; set; }
    }
}

File: FE/Mobile/SP25SE070-V3/SP25SE070-v3/src/types/report.ts
export interface ReportResponse {
  reportID: number;
  reportCode: string;
  description: string;
  imageURL: string;
  avatarOfQuestioner: string;
  avatarOfAnswer: string;
  createdDate: string;
  isTrainned: boolean;
  questionerID: number;
  questionerName: string;
  questionOfUser: string;
  answerFromExpert?: string;
  answererName: string;
  status?: "Pending" | "Answered";
}

export interface TodayTaskResponse {
  workLogId: number;
  workLogName: string;
  status: string;
  time: string;
}

export interface EmployeeProductivityResponse {
  tasksCompleted: number;
  hoursWorked: number;
  skillScore: number;
  aiReportsSubmitted: number;
  tasksPendingToday: number;
  chartData: ChartData;
}

export interface ChartData {
  tasks: Task[];
}

export interface Task {
  label: string;
  count: number;
}
